front/lib/api/files/upload.ts

```diff
@@ -502,12 +572,165 @@ export async function uploadToCloudStorage(
 
   const preProcessingRes = await maybeApplyPreProcessing(auth, file);
   if (preProcessingRes.isErr()) {
+    await file.markAsFailed();
     return new Err({
       name: "dust_error",
       code: "internal_server_error",
       message: `Failed to process the file : ${preProcessingRes.error}`,
     });
   }
 
+  // TODO(JIT) the tool output flow do not go through this path.
+  const jitEnabled = await isJITActionsEnabled(auth);
+  const isJitCompatibleUseCase = file.useCase === "conversation";
+  const hasJitRequiredMetadata =
+    file.useCase === "conversation" &&
+    !!file.useCaseMetadata &&
+    !!file.useCaseMetadata.conversationId;
+  const isJitSupportedContentType = isSupportedPlainTextContentType(
```

Comment:
It does feel you're going against the whole `PreprocessingFunction` logic by mimeType here. Wouldn't it be possible to add a wrapper on the conversation PreprocessingFunction for the mimeType we want to inject. We could have clear different wrapper for tables and documents.

The data source creation could be done before we even call the processing function no?

Reply:
I think I see what bothers you, let me look at what the changes could look like.